pkg/services/librarypanels/database.go

```diff
@@ -252,14 +253,21 @@ func (lps *LibraryPanelService) getLibraryPanel(c *models.ReqContext, uid string
 }
 
 // getAllLibraryPanels gets all library panels.
-func (lps *LibraryPanelService) getAllLibraryPanels(c *models.ReqContext) ([]LibraryPanelDTO, error) {
-	orgID := c.SignedInUser.OrgId
+func (lps *LibraryPanelService) getAllLibraryPanels(c *models.ReqContext, limit int64) ([]LibraryPanelDTO, error) {
 	libraryPanels := make([]LibraryPanelWithMeta, 0)
 	err := lps.SQLStore.WithDbSession(c.Context.Req.Context(), func(session *sqlstore.DBSession) error {
-		sql := sqlStatmentLibrayPanelDTOWithMeta + "WHERE lp.org_id=?"
-		sess := session.SQL(sql, orgID)
-		err := sess.Find(&libraryPanels)
-		if err != nil {
+		builder := sqlstore.SQLBuilder{}
+		builder.Write(sqlStatmentLibrayPanelDTOWithMeta)
+		builder.Write(" LEFT JOIN dashboard AS dashboard on lp.folder_id = dashboard.id")
```

Comment:
So the `LEFT JOIN` here does one good thing and one bad thing. 

- The good thing is that it includes all Library Panels in the General Folder
- The bad thing is that it includes all Library Panels in any non existing folder

I'm thinking the latter can be somehow prevented by checking the existence of folder when creating/updating a Library Panel and we also need to delete any library panels when folders are deleted as we don't have any foreign keys in Grafana.

Reply:
I had one thought of doing a UNION query with all Library Panels in the General folder and all Library Panels INNER JOINED to dashboards table but UNION queries aren't that performant, right?

Any other suggestions on how to solve this?

Reply:
Can you describe why this includes all library panels in general? think general folder has Id -1  , but it does not really exist in the DB if I remember correctly so this left join should not match any library panel in General

Reply:
> - The good thing is that it includes all Library Panels in the General Folder
> - The bad thing is that it includes all Library Panels in any non existing folder
Hmm, not sure how to explain this further except the LEFT JOIN takes any library panels `folder_id` and joins it with `dashboard.id` so I thought that means that any `folder_id` would be included in the end result. But it's been a while since I dabbled with SQL so I'm probably wrong.

Reply:
I think this will just join the folder (this is a 1 to 1 relationship), but for the general folder, it will not match so that will be NULL. 

![Screenshot from 2021-02-24 08-20-49](https://user-images.githubusercontent.com/10999/108962547-3b1feb00-7679-11eb-895e-9e1852ebeb94.png)

Reply:
Yes exactly, is this something that is Ok or should we do an INNER JOIN instead and UNION query with a second query that retrieves all library panels for folder_id = 0?

Reply:
Think it's ok, inner join we would not get back any in general folder

Reply:
And require a second query